product_configurator/models/product_config.py

```diff
@@ -421,7 +422,10 @@ def get_cfg_weight(self, value_ids=None, custom_vals=None):
 
         self = self.with_context(active_id=product_tmpl.id)
 
-        value_ids = flatten(value_ids)
+        # Use the updated flatten method (using itertools.chain)
+        value_ids = list(
+            chain.from_iterable(v if isinstance(v, list) else [v] for v in value_ids)
```

Comment:
Consider also other iterables (ie: tuple)

```
from collections.abc import Iterable
```

```suggestion
chain.from_iterable(v if isinstance(v, Iterable) else [v] for v in value_ids)
```